### PR TITLE
Capture command output line by line and debug output

### PR DESCRIPTION
**Pull request by k2tzumi**
refs #620

first

### CODE REVIEW

File: capturer.go [hunk 1]
```diff
@@ -41,8 +41,14 @@ type Capturer interface {
 
 	CaptureExecCommand(command, shell string, background bool)
 	CaptureExecStdin(stdin string)
+	CaptureExecStdoutStart(command string)
 	CaptureExecStdout(stdout string)
+	CaptureExecStdoutLine(text string)
```
Comment (k2tzumi) [inline capturer.go:46]: Should we integrate capture instructions?

File: capture/runbook.go [hunk 2]
```diff
@@ -449,6 +453,18 @@ func (c *cRunbook) CaptureExecStdout(stdout string) {
 	r.currentExecTestCond = append(r.currentExecTestCond, fmt.Sprintf("current.stdout == %#v", stdout))
```
Comment (k2tzumi) [inline capture/runbook.go:453]: If we integrate CaptureExecStdout and CaptureExecStdoutLine, what should we do here?

**Review by k2tzumi — COMMENTED** [review]
@k1LoW 
 I have only considered changing the runn.Capturer interface. I would like to get your opinion.
Would it be better to change the `current.stdout` from string to slice for the step execution results accordingly?

**Comment by k1LoW** [comment]
I don't have a specific implementation design in mind, but I hope to be able to make it an io.Writer instead of a string.
It would be a big change, but I believe it would be well worth it.

File: testdata/exec.yml.debugger.golden [hunk 3]
```diff
@@ -1,37 +1,67 @@
...
+  echo "$i second elapsed"
+  sleep 1
+done)<<<<<
 -----START COMMAND-----
 sleep 1000
 -----END COMMAND-----
+>>>>>START EXEC STDOUT (sleep 1000)>>>>>
+<<<<<END EXEC STDOUT (sleep 1000)<<<<<
+>>>>>START EXEC STDERR (sleep 1000)>>>>>
+<<<<<END EXEC STDERR (sleep 1000)<<<<<
```
Comment (k2tzumi) [inline testdata/exec.yml.debugger.golden:67]: It may or may not be output depending on the timing?

File: exec.go [hunk 4]
```diff
@@ -80,38 +86,130 @@ func (rnr *execRunner) run(ctx context.Context, c *execCommand, s *step) error {
...
+	if err != nil {
+		return fmt.Errorf("error starting command: %w", err)
+	}
+
+	done := make(chan execResult)
+	go func() {
+		var sob, seb bytes.Buffer
+		scanner := bufio.NewScanner(io.TeeReader(stdout, io.MultiWriter(&sob, io.Discard)))
+		o.capturers.captureExecStdoutStart(c.command)
+		for scanner.Scan() {
```
Comment (k1LoW) [inline exec.go:112]: If I output line by line, will it be broken if I use control characters?

Comment (k2tzumi) [reply exec.go:112]: Not all patterns have been verified, though.
```
go run cmd/runn/main.go run --scopes run:exec  --debug testdata/book/exec.yml
Run "exec" on "Exec test".steps[0]
-----START COMMAND-----
echo hello world!!
-----END COMMAND-----
>>>>>START EXEC STDOUT (echo hello world!!)>>>>>
hello world!!
<<<<<END EXEC STDOUT (echo hello world!!)<<<<<
>>>>>START EXEC STDERR (echo hello world!!)>>>>>
<<<<<END EXEC STDERR (echo hello world!!)<<<<<
Run "test" on "Exec test".steps[0]

Run "exec" on "Exec test".steps[1]
-----START COMMAND-----
cat
-----END COMMAND-----
-----START STDIN-----
hello world!!

-----END STDIN-----
>>>>>START EXEC STDOUT (cat)>>>>>
hello world!!
<<<<<END EXEC STDOUT (cat)<<<<<
>>>>>START EXEC STDERR (cat)>>>>>
<<<<<END EXEC STDERR (cat)<<<<<
Run "test" on "Exec test".steps[1]

Run "exec" on "Exec test".steps[2]
-----START COMMAND-----
basename `echo $0`
-----END COMMAND-----
>>>>>START EXEC STDOUT (basename `echo $0`)>>>>>
bash
<<<<<END EXEC STDOUT (basename `echo $0`)<<<<<
>>>>>START EXEC STDERR (basename `echo $0`)>>>>>
<<<<<END EXEC STDERR (basename `echo $0`)<<<<<
Run "test" on "Exec test".steps[2]

Run "exec" on "Exec test".steps[3]
-----START COMMAND-----
for i in {1..5}
do
  echo -e "$i second\telapsed\07$i"
  sleep 1
done
-----END COMMAND-----
>>>>>START EXEC STDOUT (for i in {1..5}
do
  echo -e "$i second\telapsed\07$i"
  sleep 1
done)>>>>>
1 second        elapsed9
2 second        elapsed:
3 second        elapsed;
4 second        elapsed<
5 second        elapsed=
<<<<<END EXEC STDOUT (for i in {1..5}
do
  echo -e "$i second\telapsed\07$i"
  sleep 1
done)<<<<<
>>>>>START EXEC STDERR (for i in {1..5}
do
  echo -e "$i second\telapsed\07$i"
  sleep 1
done)>>>>>
<<<<<END EXEC STDERR (for i in {1..5}
do
  echo -e "$i second\telapsed\07$i"
  sleep 1
done)<<<<<
Run "test" on "Exec test".steps[3]

Run "exec" on "Exec test".steps[4]
-----START COMMAND-----
sleep 1000
-----END COMMAND-----
Run "test" on "Exec test".steps[4]
>>>>>START EXEC STDOUT (sleep 1000)>>>>>
.

1 scenario, 0 skipped, 0 failures
<<<<<END EXEC STDOUT (sleep 1000)<<<<<
```

**Pull request closed without merging**